### PR TITLE
DASH-1120 Wallet details

### PR DESCRIPTION
**Pull request by trukhilio**

### CODE REVIEW

File: client/src/hooks/governance.ts [hunk 1]
```diff
@@ -115,11 +123,55 @@ export const useGetCandidates = (
...
+
+      const { fields } = jiraCandidatesResults[0] || {};
+
+      const candidatesVotesRersults = {
+        currentBoardVotingPower: fields?.customfield_10183,
+        boardVotingPowerLastUpdate: fields?.customfield_10184,
+        candidatesIdsList: fields?.issuelinks
+          .filter(issueLink => issueLink?.type?.outward === 'votes on')
+          .map(({ outwardIssue: { key } }) => key?.replace('FB-', '')),
+      };
```
Comment (andreyvEze) [inline client/src/hooks/governance.ts:160]: const for `customfield_10184`, `votes on`, `FB-`?

File: client/src/pages/GovernancePage/components/InfoBadgeComponent/InfoBadgeComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  title: string;
+  type: string;
+};
+
+export const InfoBadgeComponent: React.FC<Props> = ({
+  message,
+  title,
+  type,
+}) => {
+  return (
```
Comment (andreyvEze) [inline client/src/pages/GovernancePage/components/InfoBadgeComponent/InfoBadgeComponent.tsx:18]: remove return

File: client/src/pages/GovernancePage/components/WalletPower/WalletPower.tsx [hunk 3]
```diff
@@ -16,19 +16,15 @@ export const WalletPower: FC<WalletPowerProps> = ({
...
-  );
-};
+}) => (
+  <p className={classnames(classes.powerValue, className)}>
+    {withLabel && (
+      <span className={classes.powerValueLabel}>{label}:&nbsp;</span>
+    )}
+    <span>
+      {power?.toString()}
+      &nbsp;
```
Comment (andreyvEze) [inline client/src/pages/GovernancePage/components/WalletPower/WalletPower.tsx:26]: should we change to `{' '}` like in other places?

File: client/src/pages/GovernancePage/components/WalletVoteStatus/WalletVoteStatus.tsx [hunk 4]
```diff
@@ -1,40 +1,64 @@
...
+          placement="top-end"
+          delay={1000}
+          rootClose={false}
+          defaultShow={false}
+          overlay={
+            <Tooltip id="question" className={classes.infoTooltip}>
+              <span>
+                If you proxy your tokens, they count towards your proxy's vote,
+                not your own. Staked tokens may be automatically proxied. To
+                stop proxying,&nbsp;
```
Comment (andreyvEze) [inline client/src/pages/GovernancePage/components/WalletVoteStatus/WalletVoteStatus.tsx:46]: `{' '}`

File: client/src/pages/GovernancePage/components/WalletVotesDetailsModal/components/MyVoteDetails/MyVoteDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  lastUpdated?: string;
+};
+
+export const MyVoteDetails: FC<Props> = ({
+  powerLabelSuffix,
+  power,
+  lastVote,
+  lastUpdated,
+}) => {
+  return (
```
Comment (andreyvEze) [inline client/src/pages/GovernancePage/components/WalletVotesDetailsModal/components/MyVoteDetails/MyVoteDetails.tsx:28]: we can remove `return`

File: client/src/pages/GovernancePage/components/WalletVotesDetailsModal/components/MyVoteDetails/MyVoteDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+              {voteFormatDate(new Date(lastVote))}
+            </span>
+          </p>
+        )}
+        <p className={classes.detailsItem}>
+          <span className={classes.detailsLabel}>
+            Current&nbsp;{powerLabelSuffix ? <>Board&nbsp;</> : ''}Voting Power:
+          </span>
+          <span className={classes.detailsValue}>
+            {power?.toLocaleString('en', {
```
Comment (andreyvEze) [inline client/src/pages/GovernancePage/components/WalletVotesDetailsModal/components/MyVoteDetails/MyVoteDetails.tsx:44]: `power` is string

File: client/src/util/general.ts [hunk 7]
```diff
@@ -316,36 +316,69 @@ export const convertToNewDate = (
...
+    voteDate =
+      dates.find(date => date > now) ||
+      new Date(dates[0].setFullYear(dates[0].getFullYear() + 1));
+  }
 
-  const nextDate =
-    dates.find(date => date > now) ||
-    new Date(dates[0].setFullYear(dates[0].getFullYear() + 1));
+  if (returnDefaultFormat)
+    return voteDate.toLocaleString('en-US', { timeZone: 'America/New_York' });
```
Comment (andreyvEze) [inline client/src/util/general.ts:351]: let's put tz into const here in the util

**Review by andreyvEze — APPROVED** [review]

File: client/src/pages/GovernancePage/components/InfoBadgeComponent/InfoBadgeComponent.tsx (hunk 2 above)
Comment (trukhilio) [reply client/src/pages/GovernancePage/components/InfoBadgeComponent/InfoBadgeComponent.tsx:18]: Done

File: client/src/pages/GovernancePage/components/WalletPower/WalletPower.tsx (hunk 3 above)
Comment (trukhilio) [reply client/src/pages/GovernancePage/components/WalletPower/WalletPower.tsx:26]: Done

File: client/src/pages/GovernancePage/components/WalletVoteStatus/WalletVoteStatus.tsx (hunk 4 above)
Comment (trukhilio) [reply client/src/pages/GovernancePage/components/WalletVoteStatus/WalletVoteStatus.tsx:46]: Done

File: client/src/pages/GovernancePage/components/WalletVotesDetailsModal/components/MyVoteDetails/MyVoteDetails.tsx (hunk 5 above)
Comment (trukhilio) [reply client/src/pages/GovernancePage/components/WalletVotesDetailsModal/components/MyVoteDetails/MyVoteDetails.tsx:28]: Done

File: client/src/pages/GovernancePage/components/WalletVotesDetailsModal/components/MyVoteDetails/MyVoteDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+          <p className={classes.detailsItem}>
+            <span className={classes.detailsLabel}>Last Board Vote:</span>
+            <span className={classes.detailsValue}>
+              {voteFormatDate(new Date(lastVote))}
+            </span>
+          </p>
+        )}
+        <p className={classes.detailsItem}>
+          <span className={classes.detailsLabel}>
+            Current&nbsp;{powerLabelSuffix ? <>Board&nbsp;</> : ''}Voting Power:
```
[the remark this answers is not in the document]
Comment (trukhilio) [reply client/src/pages/GovernancePage/components/WalletVotesDetailsModal/components/MyVoteDetails/MyVoteDetails.tsx:41]: Done

File: client/src/util/general.ts (hunk 7 above)
Comment (trukhilio) [reply client/src/util/general.ts:351]: Done

**Pull request merged**